client/src/components/App/Content/InvestigationForm/InvestigationForm.tsx

```diff
@@ -81,38 +70,68 @@ const InvestigationForm: React.FC = (): JSX.Element => {
+            if(isInvestigationValid()){
+                confirmFinishInvestigation(epidemiologyNumber);
+            } else {
+                Swal.fire({
+                    icon: 'error',
+                    title: 'חלק מן השדות אינם תקניים, נא מלא אותם מחדש ונסה שוב.'
+                });
+            }
+        } else {
+            setNextTab(currentTab+1);
+        }
+    }
 
-    const shouldDisableButton = isButtonDisabled(currentTab.name);
     return (
         <div className={classes.content}>
             <ExposureAndFlightsContextProvider value={exposuresAndFlightsVariables}>
-                <PersonalInfoContextProvider value={personalInfoValue}>
                     <ClinicalDetailsDataContextProvider value={clinicalDetailsVariables}>
                         <StartInvestigationDateVariablesProvider value={startInvestigationDateVariables}>
                             <InvestigationInfoBar
-                                onExitInvestigation={saveCurrentTab}
+                                currentTab = {currentTab}
                             />
                                 <div className={classes.interactiveForm}>
                                     <TabManagement
-                                        currentTab={currentTab}
-                                        setCurrentTab={setCurrentTab}
-                                        onTabClicked={() => shouldDisableButton ? setShowSnackbar(true) : saveCurrentTab()}
-                                        shouldDisableChangeTab={shouldDisableButton}
+                                        currentTab = {currentTab}
+                                        moveToNextTab = {moveToNextTab}
+                                        setCurrentTab = {setCurrentTab}
+                                        setNextTab = {setNextTab}
                                     />
                                     <div className={classes.buttonSection}>
-                                        <PrimaryButton test-id={currentTab.id === LAST_TAB_ID ? 'endInvestigation' : 'continueToNextStage'}
-                                            onClick={() => {
-                                                currentTab.id === LAST_TAB_ID ? confirmFinishInvestigation(epidemiologyNumber) : handleSwitchTab();
-                                            }}
-                                            disabled={shouldDisableButton}>
-                                           {currentTab.id === LAST_TAB_ID ? END_INVESTIGATION : CONTINUE_TO_NEXT_TAB}
+                                        <PrimaryButton 
+                                            type="submit"
+                                            form={`form-${currentTab}`}
+                                            test-id={currentTab === LAST_TAB_ID ? 'endInvestigation' : 'continueToNextStage'}
+                                            onClick={handleClick}
+                                            disabled={false}>
```

Comment:
Why is it hardcoded to be false? Don't you want it as a variable that you can control?

Reply:
You can move tab even if the it's not valid, But you can't finish investigation if it's invalid

Reply:
Ok but the button will always be enabled that way

Reply:
I'll ask Elad about whether to disabled the button or not

Reply:
i will delete it, it useless